### ISSUE TITLE
Live TV streams broken

### ISSUE THREAD

**Issue by glorang** [glorang/NUPlayer#8]
Sunday 18 or Monday 19 July something changed in the VRT live streams that NUPlayer cannot handle. 

Either the video activity crashes or video stops for a few seconds while audio continues making watching Live TV impossible.

Feel free to comment on this issue if you have any relevant feedback.

**Comment by michaelarnauts** [comment]
I think it might be due to this: https://github.com/add-ons/plugin.video.vrt.nu/pull/893

**Comment by mediaminister** [comment]
VRT removed the HLS AES livestreams: https://www.vrt.be/heb-je-een-vraag/s/article/Waarom-werkt-sinds-19-juli-2021-de-stream-van-de-VRT-niet-meer-op-mijn-toestel

You can now only use the MPEG-DASH live streams with Widevine DRM ( `mpeg_dash` targetUrl)

**Comment by glorang** [comment]
Thanks for the info guys, much appreciated. 

NUPlayer was always using the DASH stream already and DRM is (always) added when required - it checks for `drm` tag in f `vualto-video-aggregator-web/rest/external/v2/videos` result. 

For NUPlayer two problems popped up since they changed the live stream:

**- Video playback is aborted and you return automatically to the home screen**

The DASH manifest is sometimes missing the 720p stream while it is still "announced" somewhere else - you can see this on the website as well, try refreshing a few times on the live stream, sometimes 720p is available, sometimes it's not. 

This is causing an `IndexOutOfBoundsException` which will crash the player. I hope VRT will fix this at some point in time.

```
E/LoadTask: Unexpected exception handling load completed
      java.lang.IndexOutOfBoundsException: Index: 6, Size: 6
        at java.util.ArrayList.get(ArrayList.java:437)
        at com.google.android.exoplayer2.source.dash.DefaultDashChunkSource.updateManifest(DefaultDashChunkSource.java:246)
        at com.google.android.exoplayer2.source.dash.DashMediaPeriod.updateManifest(DashMediaPeriod.java:161)
        at com.google.android.exoplayer2.source.dash.DashMediaSource.processManifest(DashMediaSource.java:898)
        at com.google.android.exoplayer2.source.dash.DashMediaSource.onManifestLoadCompleted(DashMediaSource.java:741)
        at com.google.android.exoplayer2.source.dash.DashMediaSource$ManifestCallback.onLoadCompleted(DashMediaSource.java:1376)
        at com.google.android.exoplayer2.source.dash.DashMediaSource$ManifestCallback.onLoadCompleted(DashMediaSource.java:1371)
        at com.google.android.exoplayer2.upstream.Loader$LoadTask.handleMessage(Loader.java:483)
        at android.os.Handler.dispatchMessage(Handler.java:106)
        at android.os.Looper.loop(Looper.java:193)
        at android.os.HandlerThread.run(HandlerThread.java:65)
```

**- Video playback is choppy**

It looks like 1080p / 50 fps without DRM works great  but with DRM / Widevine there seems to be performance issues and a lot of dropped frames are observed - See issue 6348 in ExoPlayer repo. Looks like they have some experimental code ready to improve this, but this should be investigated which needs time.

I've worked around both issues temporarily by setting the max video size to 540p. This is far from ideal but better than a crashing or choppy player I guess....

**Comment by glorang** [comment]
@davidvde72 @bmesuere @michaelarnauts I could use some testing validation on this one.

As you might have noticed the 1080p streams have disappeared from latest release (1.3.5) due to the problems described in this issue and everything plays at 540p max at the moment.

I've now released 1.3.6 in the beta channel that allows you to select the video quality from the video player via the gear icon.

From my testing it seems that:

- Canvas and Ketnet streams both work flawlessly at 1080p (and 720p)
- Een stream has any of the following issues:
  - Is missing the 720p stream from time to time
  - When 720p is available the playback is choppy (audio continues but video doesn't)
  - Selecting 720p sometimes results in the 1080p stream (see Info overlay)
  - 1080p always crashes after 2nd media chunk (20-40 seconds of playback time approximately) and you return to the Home screen

Could you test around a bit with the different video qualities on the different streams and let me know if you observe anything different ? I'm assuming your results will be similar but it would be good to check on different hardware and different internet connection etc.

For info; for those testing purposes the stream info overlay is highly useful, enable developer mode in Settings first, then you can toggle the stream info overlay from the player with the "Info" icon.

Many thanks,

**Comment by bmesuere** [comment]
Sorry, I was on vacation for the last week. I just did some tests on my Chromecast with Google TV:

- een: 720p not listed, 1080p plays but video often stops (audio continues) it doesn't crash like you reported.
- canvas: all streams listed, everything seems to work
- Ketnet: all streams listed, everything seems to work

Update een: if the 720p stream is listed, it also results in choppy playback as with the 1080 stream

**Comment by glorang** [comment]
Ok great, thanks for testing! Looks like VRT fixed the "één" stream and now it also works correctly again in NUPlayer. I'll try to push out an update that selects 1080p automatically as soon as possible.

**Comment by glorang** [comment]
Fixed in 1.3.7. Available now in Beta and Production channels on Google Play.
